ndb/src/google/cloud/ndb/_ports.py

```diff
@@ -0,0 +1,186 @@
+
+        Args:
+            latitude (Union[float, str]): The latitude value to be converted.
+                Must be in the range ``[-90, 90]``.
+            longitude (Union[float, str]): The longitude value to be converted.
+                Must be in the range ``[-180, 180]``.
+
+        Returns:
+            Tuple[float, float]: The components of the comma-separated ``value``.
+
+        Raises:
+            .BadValueError: If one of ``latitude``, ``longitude`` is not
+                convertible to a floating point number.
+            .BadValueError: If ``latitude`` is outside the range ``[-90, 90]``.
+            .BadValueError: If ``longitude`` is outside the range
+                ``[-180, 180]``.
+        """
+        try:
+            latitude = float(latitude)
+            longitude = float(longitude)
+        except (TypeError, ValueError):
+            template = (
+                "Expected floats for lat and long; received {} (a {}) "
+                "and {} (a {})."
+            )
+            msg = template.format(
+                latitude,
+                latitude.__class__.__name__,
+                longitude,
+                longitude.__class__.__name__,
+            )
+            raise exceptions.BadValueError(msg)
+
+        if abs(latitude) > 90:
+            raise exceptions.BadValueError(
+                "Latitude must be between -90 and 90; "
+                "received {:f}".format(latitude)
+            )
+
+        if abs(longitude) > 180:
+            raise exceptions.BadValueError(
+                "Longitude must be between -180 and 180; "
+                "received {:f}".format(longitude)
+            )
+
+        return latitude, longitude
+
+    def __eq__(self, other):
+        if not isinstance(other, GeoPt):
+            try:
+                other = GeoPt(other)
+            except exceptions.BadValueError:
+                return NotImplemented
+
+        return self.lat == other.lat and self.lon == other.lon
+
+    def __lt__(self, other):
+        if not isinstance(other, GeoPt):
+            try:
+                other = GeoPt(other)
+            except exceptions.BadValueError:
+                return NotImplemented
+
+        return (self.lat, self.lon) < (other.lat, other.lon)
+
+    def __hash__(self):
+        """Returns an integer hash of this point.
+
+        Implements Python's hash protocol so that GeoPts may be used in sets
+        and as dictionary keys.
+
+        Returns:
+            int: The hash of the ``(latitude, longitude)`` tuple.
+        """
+        return hash((self.lat, self.lon))
+
+    def __repr__(self):
+        """Returns an ``eval()``-able string representation of this GeoPt.
+
+        The returned string is of the form
+        ``datastore_types.GeoPt(latitude, longitude)``.
+
+        Returns:
+            str: The repr for this instance.
+        """
+        return "datastore_types.GeoPt({!r}, {!r})".format(self.lat, self.lon)
```

Comment:
No longer in `datastore_types`. Prefix not necessary anyway.

Reply:
Do you think it's necessary to preserve the original?